torch/distributed/algorithms/ddp_comm_hooks/post_localSGD_hook.py

```diff
@@ -70,6 +72,8 @@ def post_localSGD_hook(
         >>> state = PostLocalSGDState(process_group=process_group, subgroup=subgroup,
                                   start_localSGD_iter=10)
         >>> ddp_model.register_comm_hook(state, post_localSGD_hook)
+        >>> # Also need to establish a model averaging module and run model averaging after ``optimizer.step()``.
```

Comment:
Maybe remove the example from here all together and in the docs above just mention that look at the docs of model_averaging for an example. I think that might be more cleaner.

Reply:
Eventually there will be multiple `averager` classes corresponding to different model averaging algorithms, but probably we will only have a single comm hook for the post-localSGD paradigm. Therefore, if we merge this to `PeriodicAverager`, we will also need to merge it to any new `Averager` class.